### PR TITLE
Use comment matcher in CXX frontend

### PR DESCRIPTION
**Pull request by KuechA**
Small improvement to comment matcher and use it in the CXX frontend.

### CODE REVIEW

File: cpg-language-cxx/src/test/kotlin/de/fraunhofer/aisec/cpg/helpers/CommentsTest.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import de.fraunhofer.aisec.cpg.graph.*
+import java.io.File
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+
+class CommentsTest {
+    @Test
+    fun testCommentMatcher() {
+        val file = File("src/test/resources/comments.c")
```
Comment (oxisto) [inline cpg-language-cxx/src/test/kotlin/de/fraunhofer/aisec/cpg/helpers/CommentsTest.kt:40]: can you move this file into the `c` subfolder? we are trying to categorise at least new files a little bit by (sub)language.

File: cpg-language-cxx/src/test/kotlin/de/fraunhofer/aisec/cpg/helpers/CommentsTest.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        val tu =
+            TestUtils.analyzeAndGetFirstTU(listOf(file), file.parentFile.toPath(), true) {
+                it.registerLanguage<CPPLanguage>()
+                it.registerLanguage<CLanguage>()
+            }
+        assertNotNull(tu)
+        val functionDoc = tu.functions["functionDoc"]
+        assertNotNull(functionDoc)
+        assertEquals("/**\n * Documentation comment\n */", functionDoc.comment)
+        val functionMultiline = tu.functions["functionMultiline"]
```
Comment (oxisto) [inline cpg-language-cxx/src/test/kotlin/de/fraunhofer/aisec/cpg/helpers/CommentsTest.kt:50]: nit: line break before the individual "vals" for readability.

File: cpg-language-cxx/src/test/resources/comments.c [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+/*
+ * Multi-line
+ * comment
+ */
+void functionMultiline() {
+  return;
+}
+
+// Simple comment
+
```
Comment (oxisto) [inline cpg-language-cxx/src/test/resources/comments.c:17]: Is this newline in between intentional?

Comment (KuechA) [reply cpg-language-cxx/src/test/resources/comments.c:17]: Yes, it is. The comment matcher should still identify it even with the additional newline between the comment and the function

**Review by oxisto — APPROVED** [review]

**Comment by KuechA** [comment]
@oxisto  The "old" `setComments` method is still used in various places. I'm not sure what to do with it. Should we remove it? It could lead to adding some comments twice now.

**Comment by oxisto** [comment]
> @oxisto The "old" `setComments` method is still used in various places. I'm not sure what to do with it. Should we remove it? It could lead to adding some comments twice now.

You can try to remove it and see if it (doesn't) break :)

**Comment by KuechA** [comment]
> > @oxisto The "old" `setComments` method is still used in various places. I'm not sure what to do with it. Should we remove it? It could lead to adding some comments twice now.
> 
> You can try to remove it and see if it (doesn't) break :)

Seems to work

**Pull request merged**